src/components/lab/Grid.js

```diff
@@ -4,38 +4,46 @@ import { Line } from 'react-konva';
 import {
   GRID_AXES_COLOR,
   GRID_AXES_STROKE_WIDTH,
+  MEASURING_ARROW_STEP,
 } from '../../config/constants';
 
-const Grid = ({ gridWidth, gridHeight, numOfxAxisTicks }) => {
-  // divide x-axis (=gridWidth) into numOfxAxisTicks equally sized ticks, each of width xTickWidth
-  const xTickWidth = gridWidth / numOfxAxisTicks;
-  // create an array xTicksArray: in return statement below, a VERTICAL line is drawn at each of the ticks in this array
-  const xTicksArray = new Array(numOfxAxisTicks)
-    .fill()
-    .map((emptyElement, index) => (index + 1) * xTickWidth);
+const Grid = ({ gridWidth, gridHeight }) => {
+  // MEASURING_ARROW_STEP is the increment with which the measuring arrow's width is increased
+  // each MEASURING_ARROW_STEP is '100nm'
+  // we want the squares in our grid to have 100nm x 100nm dimensions
+  // for clarity, re-assign MEASURING_ARROW_STEP to another variable
+  const GRID_SQUARE_WIDTH_AND_HEIGHT = MEASURING_ARROW_STEP;
 
-  // to ensure grid is square, assign yTickHeight = xTickWidth
-  const yTickHeight = xTickWidth;
+  // in return statement below, a VERTICAL line is drawn at each of the points in xTicksArray
+  // for alignment purposes, initialize xTicksArray to contain the value (gridWidth / 2)
+  // this ensures that the vertical radiation line emitted by the charge is in line with the grid
+  const centralXTick = gridWidth / 2;
+  const xTicksArray = [centralXTick];
+  let xTickLeft = centralXTick - GRID_SQUARE_WIDTH_AND_HEIGHT;
+  while (xTickLeft >= 0) {
+    xTicksArray.unshift(xTickLeft);
+    xTickLeft -= GRID_SQUARE_WIDTH_AND_HEIGHT;
+  }
+  let xTickRight = centralXTick + GRID_SQUARE_WIDTH_AND_HEIGHT;
+  while (xTickRight <= gridWidth) {
+    xTicksArray.push(xTickRight);
+    xTickRight += GRID_SQUARE_WIDTH_AND_HEIGHT;
+  }
```

Comment:
Very interesting approach! Thanks for the suggestion. It does indeed work well with some modifications:

```
// get total number of ticks for the width
const totalNumberOfXTicks = Math.floor(gridWidth/GRID_SQUARE_WIDTH_AND_HEIGHT) + 1
...
// create x ticks array
const xTicksArray = new Array(totalNumberOfXTicks).fill().map((emptyElement, i) => startingXTick+i*GRID_SQUARE_WIDTH_AND_HEIGHT)
```

- the `+ 1` in `totalNumberOfXTicks` is necessary to prevent this:

![Screen Shot 2021-01-21 at 9 10 17 AM](https://user-images.githubusercontent.com/19311953/105322061-979a7100-5bc8-11eb-90d8-a524ae75556b.png)

- with `new Array(...).keys().map(...)` I get an error: `new Array(...).keys().map(...) is not a function`. basically `.keys()` creates an Array Iterator object, which one can't use `.map()` on.

let me know if I'm missing anything. I like the approach, will amend.

Reply:
I coded it quickly in an external editor, so as long as something is working for you, I'll go with it as well^^